### ISSUE TITLE
Permit null value

### ISSUE THREAD

**Issue by n3wtron** [civitz/viper#33]
The generated ConfigurationBean.validateProperties() method check the value nullability and raise an IllegalArgumentException if it's null.

You could create an annotation like @CdiConfiguration.ConfigValidator (e.g CdiConfiguration.ConfigOptional) used to annotate an boolean method on the configuration enum.

**Comment by civitz** [comment]
Cool! This is somewhat related to #4 , we need to make these work together in an organic setup. 

We can also consider the proposed solution for #27 , like 
```
@CdiConfiguration(missing=MissingStrategy.NONE)
@CdiConfiguration(missing=MissingStrategy.NULL) // the proposed default
@CdiConfiguration(missing=MissingStrategy.NULL_OR_EMPTY) // the current default
```

One could also argue that "presence of a value" is itself a validation: with this frame of mind, we could remove presence check altogether (and maybe provide a set of sensible validation methods).

**Comment by civitz** [comment]
If the validation argument is not meaningful, we should consider making ConfigurationResolver return an `Optional<String>` instead of a string. This would make it easier to tell missing values from present values, but using `java.util.Optional` makes it impossible to return an `Optional` with a `null` value.

Choices, choices...

**Comment by civitz** [comment]
Ok, I've reasoned about this... And the conclusion is: null checking **is** validation. Therefore we should omit null checking **AND** missing strategy.

Optional values (see #4 ) is a related issue that can be solved via a custom resolver, providing nulls or default values.

This will further simplify the generated code.

Solution to this and #4 combined will be issued soon.

The only open question is normalization (see #27), which is somehow related to validation: it also can be done in the resolver. A sane default implementation can be useful, but it may be overkill to do in this small project.

**Comment by civitz** [comment]
Closed in #34
